jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/DisplayMessageFragment.java

```diff
@@ -1,17 +1,49 @@
 package com.e_conomic.jonfirstapp;
 
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 import android.view.View;
+import android.widget.TextView;
+
 
 public class DisplayMessageFragment extends Fragment {
 
+    // The message to be displayed.
+    private String message = "";
+
+    // The TextView that displays the message.
+    private TextView display_message;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+    }
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         return inflater.inflate(R.layout.display_message, container, false);
     }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // Get the TextView that shows the message.
+        display_message = (TextView) getView().findViewById(R.id.display_message_view);
+        display_message.setText(message);
+    }
+
+    /** Updates the TextView that shows the message. */
+    public void updateMessage(String newMessage) {
+        message = newMessage;
```

Comment:
Hmm.. but I also use it in `onViewCreated` to set the message after a configuration change? :) E.g. if you turn your phone it will still show the message you entered before you turned it!